colossalai/interface/optimizer.py

```diff
@@ -58,6 +58,40 @@ def backward(self, loss: Tensor, *args, **kwargs):
     def backward_by_grad(self, tensor: Tensor, grad: Tensor):
         torch.autograd.backward(tensor, grad)
 
+    def backward_b_by_grad(self, tensors: Tensor, grad_tensors: Tensor, inputs: Tensor, retain_graph: bool = True):
+        """
+        Performs a backward pass for dx, we only calculate dx = w*dy here
+
+        Args:
+            tensor (Tensor): y or loss of current chunk;
+            grad_tensors (Tensor): dy of current chunk;
+            input_obj (Tensor): x of current chunk;
+            retain_graph (bool): default to be True, we retain graph in backward_b
+        """
+        torch.autograd.backward(
+            tensors=tensors,
+            grad_tensors=grad_tensors,
+            inputs=inputs,
+            retain_graph=retain_graph,
+        )
+
+    def backward_w_by_grad(self, tensors: Tensor, grad_tensors: Tensor, inputs: Tensor, retain_graph: bool = False):
+        """
+        Performs a backward pass for dw, we only calculate dw = x*dy here
+
+        Args:
+            tensor (Tensor): y or loss of current chunk;
+            grad_tensors (Tensor): dy of current chunk;
+            input_obj (Tensor): w;
+            retain_graph (bool): default to be False, we release graph in backward_w
+        """
+        torch.autograd.backward(
+            tensors=tensors,
+            grad_tensors=grad_tensors,
+            inputs=inputs,
+            retain_graph=retain_graph,
+        )
```

Comment:
These three methods are too similar. Just leave one and pass difference arguments when calling them.

Reply:
> These three methods are too similar. Just leave one and pass difference arguments when calling them.

Use on func "backward_b_w_by_grad" in optimizer instead. Should we change the name ?